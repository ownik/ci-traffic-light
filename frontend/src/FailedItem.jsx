import React from 'react';
import PropTypes from 'prop-types';
import ProgressSpinner from './ProgressSpinner';

const FailedItem = ({ displayName, href, running }) => (
  <li>
    <span>
      {!href || href === '' ? displayName : <a href={href}>{displayName}</a>}
    </span>
    {running && <ProgressSpinner />}
  </li>
);

FailedItem.propTypes = {
  displayName: PropTypes.string,
  href: PropTypes.string,
  running: PropTypes.bool,
};

FailedItem.defaultProps = {
  href: '',
};

export default FailedItem;
