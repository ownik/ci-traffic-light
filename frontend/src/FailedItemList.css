div.failed-item-list {
  visibility: visible;
  display: block;
  color: white;
  font-size: 4em;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  max-width: 50%;
}

div.failed-item-list > ul {
  list-style: none;
}

div.failed-item-list > ul > li {
  transition: opacity 1s;
  display: flex;
  align-items: center;
}

div.failed-item-list > ul > li.enter {
  opacity: 0;
}

div.failed-item-list > ul > li.enter-active {
  opacity: 1;
}

div.failed-item-list > ul > li.exit {
  opacity: 1;
}

div.failed-item-list > ul > li.exit-active {
  opacity: 0;
}

div.failed-item-list > ul > li > a,
a:visited {
  display: inline-block;
  line-height: 0.85;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  max-width: 100%;
  margin-bottom: 17px;
}

div.failed-item-list > ul > li > a:link,
a:visited {
  color: white;
  text-decoration: none;
}

div.failed-item-list > ul > li > a:hover,
a:active {
  color: white;
  text-decoration: none;
  border-bottom: 2px solid white;
  margin-bottom: 15px;
}
