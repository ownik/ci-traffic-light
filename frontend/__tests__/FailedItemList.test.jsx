import React from 'react';
import { shallow } from 'enzyme';
import FailedItemList from '../src/FailedItemList';
import FailedItem from '../src/FailedItem';

describe('FailedItemList', () => {
  test('renders without crashing', () => {
    shallow(<FailedItemList />);
  });

  test('check className', () => {
    expect(shallow(<FailedItemList />).props()).toHaveProperty(
      'className',
      'failed-item-list'
    );
  });

  const items = [
    { displayName: 'item1', href: 'testhref1' },
    { displayName: 'item2', href: 'testhref2' },
    { displayName: 'item3' },
  ];

  test('check display items', () => {
    const wrapper = shallow(<FailedItemList items={items} />);
    expect(wrapper.find(FailedItem)).toHaveLength(items.length);
    let actual = [];
    wrapper.find(FailedItem).forEach((item) => {
      actual.push(item.props());
    });
    expect(actual).toStrictEqual([
      { displayName: 'item1', href: 'testhref1' },
      { displayName: 'item2', href: 'testhref2' },
      { displayName: 'item3', href: '' },
    ]);
  });

  test('check display items snapshot', () => {
    const wrapper = shallow(<FailedItemList items={items} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
