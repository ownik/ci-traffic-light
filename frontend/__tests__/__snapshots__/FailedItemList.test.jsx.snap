// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FailedItemList check display items snapshot 1`] = `"<div class=\\"failed-item-list\\"><ul><li><a href=\\"testhref1\\">item1</a></li><li><a href=\\"testhref2\\">item2</a></li><li>item3</li></ul></div>"`;
